sdk/storage/azure-storage-blob/azure/storage/blob/__init__.py

```diff
@@ -81,3 +85,73 @@
     'ResourceTypes',
     'AccountPermissions',
 ]
+
+
+def upload_blob_to_url(
+        blob_url,  # type: str
+        data,  # type: Union[Iterable[AnyStr], IO[AnyStr]]
+        overwrite=False,  # type: bool
+        max_connections=1,  # type: int
+        encoding='UTF-8', # type: str
+        credential=None,  # type: Any
+        **kwargs):
+    """Upload data to a given URL
+
+    The data will be uploaded as a block blob.
+
+    :param str blob_url:
+        The full URI to the blob. This can also include a SAS token.
+    :param data:
+        The data to upload. This can be bytes, text, an iterable or a file-like object.
+    :type data: bytes or str or Iterable
+    :param bool overwrite:
+        Whether the blob to be uploaded should overwrite the current data.
+        If True, upload_blob_to_url will silently overwrite the existing data. If set to False, the
+        operation will fail with a ResourceExistsError.
+    :param credential:
+        The credentials with which to authenticate. This is optional if the
+        blob URL already has a SAS token. The value can be a SAS token string, an account
+        shared access key, or an instance of a TokenCredentials class from azure.identity.
+    :returns: Blob-updated property dict (Etag and last modified)
+    :rtype: dict[str, Any]
+    """
+    with BlobClient(blob_url, credential=credential) as client:
+        return client.upload_blob(
+            data=data,
+            blob_type=BlobType.BlockBlob,
+            overwrite=overwrite,
+            max_connections=max_connections,
+            encoding=encoding,
+            **kwargs)
+
+
+def download_blob_from_url(
+        blob_url,  # type: str
+        file_path,  # type: str
```

Comment:
It would be neat to be able to provide a file-like object (e.g. [`StringIO`](https://docs.python.org/2/library/stringio.html#StringIO.StringIO)) when calling the method. Which would be easy to add, but the argument name may be slightly misleading if we do. Thoughts?